src/tls.c

```diff
@@ -1005,7 +986,8 @@ static ssize_t connTLSSyncWrite(connection *conn_, char *ptr, ssize_t size, long
 
     setBlockingTimeout(conn, timeout);
     SSL_clear_mode(conn->ssl, SSL_MODE_ENABLE_PARTIAL_WRITE);
-    int ret = SSL_write(conn->ssl, ptr, size);
+    ERR_clear_error();
+    int ret = updateStateAfterSSLIO(conn, SSL_write(conn->ssl, ptr, size), 0);
```

Comment:
personally, i'd rather not hide the SSL_Write (main purpose of this function) inside the same line of the state update.